### DISCUSSION TITLE
Rule for __dirname

### DISCUSSION THREAD

**Discussion by kibertoad** [biomejs/biome#4245] [Rule suggestion]
Would it make sense to have a rule that would check for the usage of __dirname? It doesn't work in ESM context, but is not easily caught by tests, as e. g. `vitest` injects `__dirname`, so it works correctly there, but then fails in production.

**Comment by dyc3** [comment]
Yeah this would be great to have. Are there any eslint rules (even from plugins) that cover this? `__filename` is also in a similar situation. I'd suggest `noLegacyPathGlobals` or `noLegacyCurrentFileGlobals` for the rule name. Feel free to file a task for it.

**Comment by kibertoad** [comment]
@ematipico Why was https://github.com/biomejs/biome/issues/4331 closed? There literally was a request to open a task for this.

**Reply by ematipico** [reply]
Because we don't know who is going to work on the task, or there aren't instructions for contributors, and we don't know how we are going to name the rule. @dyc3 proposed some, but we haven't agreed to any. 

If we want to open a task, we should provide all the info for a contributor to pick up the task and work on it. I will re-open it, but please let's provide more information for contributors.

**Reply by kibertoad** [reply]
@ematipico Thank you. Can you give a good example of how it needs to be described?

**Reply by ematipico** [reply]
https://github.com/biomejs/biome/issues/4333

**Comment by Netail** [comment]
Seems to be implemented: https://biomejs.dev/linter/rules/no-global-dirname-filename/ :)
